src/library_browser.js

```diff
@@ -766,57 +766,81 @@ var LibraryBrowser = {
     safeSetTimeout(() => _emscripten_run_script(script), millis);
   },
 
-  // TODO: currently not callable from a pthread, but immediately calls onerror() if not on main thread.
-  emscripten_async_load_script__deps: ['$UTF8ToString'],
-  emscripten_async_load_script: (url, onload, onerror) => {
+  emscripten_async_run_script__deps: ['emscripten_run_script', '$safeSetTimeout'],
+  emscripten_async_run_script: (script, millis) => {
+    // TODO: cache these to avoid generating garbage
+    safeSetTimeout(() => _emscripten_run_script(script), millis);
+  },
```

Comment:
It looks like these 5 lines got duplicated.

Reply:
Removed the duplicate code @sbc100

Reply:
Just want to be sure whether I am heading in the right direction fundamentally for the api unification